src/main/java/ru/svgogin/service/spark/web/SparkController.java

```diff
@@ -33,26 +36,36 @@ public SparkController(SparkService sparkService) {
   }
 
   @GetMapping("/{inn}")
-  public ResponseEntity<CompanyDto> getCompanyByInn(@PathVariable("inn") String inn) {
+  public ResponseEntity<CompanyDto> getCompanyByInn(@PathVariable("inn")
+                                                      @Pattern(regexp = "^(\\d{10}|\\d{12})$",
+                                                               message = "Inn should contain "
+                                                                         + "10 or 12 digits")
+                                                          String inn) {
     log.info("getCompanyByInn " + inn);
     return new ResponseEntity<>(sparkService.findByInn(inn), HttpStatus.OK);
   }
 
   @PostMapping()
-  public ResponseEntity<CompanyDto> saveCompany(@RequestBody CompanyDto companyDto) {
+  public ResponseEntity<CompanyDto> saveCompany(@Valid @RequestBody CompanyDto companyDto) {
     log.info("saveCompany with inn " + companyDto.getInn());
     return new ResponseEntity<>(sparkService.save(companyDto), HttpStatus.OK);
   }
 
   @PutMapping("/{inn}")
-  public ResponseEntity<CompanyDto> updateCompany(@PathVariable("inn") String inn,
-                                                  @RequestBody CompanyDto companyDto) {
+  public ResponseEntity<CompanyDto> updateCompany(@PathVariable("inn")
+                                                    @Pattern(regexp = "^(\\d{10}|\\d{12})$",
+                                                             message = "Inn should contain "
+                                                        + "10 or 12 digits") String inn,
+                                                  @Valid @RequestBody CompanyDto companyDto) {
     log.info("updateCompany " + companyDto.getInn());
     return new ResponseEntity<>(sparkService.update(inn, companyDto), HttpStatus.OK);
   }
 
   @DeleteMapping("/{inn}")
-  public ResponseEntity<CompanyDto> deleteCompany(@PathVariable("inn") String inn) {
+  public ResponseEntity<CompanyDto> deleteCompany(@PathVariable("inn")
+                                                    @Pattern(regexp = "^(\\d{10}|\\d{12})$",
+                                                             message = "Inn should contain "
+                                                        + "10 or 12 digits") String inn) {
```

Comment:
It is better to format this way (and in another methods)
```suggestion
  public ResponseEntity<CompanyDto> deleteCompany(
      @PathVariable("inn")
      @Pattern(regexp = "^(\\d{10}|\\d{12})$", message = "Inn should contain 10 or 12 digits") String inn
      ) {
```

Reply:
[INFO] --- maven-checkstyle-plugin:3.1.1:check (validate) @ spark ---
[INFO] Starting audit...
[ERROR] D:\ServiceRepository\src\main\java\ru\svgogin\service\spark\web\SparkController.java:66: Line is longer than 100 characters (found 154). [LineLength]
Audit done.

Reply:
Done